rethinkdb/tests/unit/test_metrics.py

```diff
@@ -0,0 +1,97 @@
+# (C) Datadog, Inc. 2020-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+"""
+Unit tests for metrics that are hard to test using integration tests, eg. because they depend on cluster dynamics.
+"""
+import pytest
+
+from datadog_checks.rethinkdb.metrics.system_jobs import collect_system_jobs
+from datadog_checks.rethinkdb.queries import QueryEngine
+from datadog_checks.rethinkdb.types import BackfillJob, IndexConstructionJob
+
+from ..utils import MockConnection
+
+pytestmark = pytest.mark.unit
+
+
+def test_jobs_metrics():
+    # type: () -> None
+    """
+    Verify jobs metrics submitted by RethinkDB are processed correctly.
+
+    We provide unit tests for these metrics because testing them in a live environment is tricky.
```

Comment:
Could you comment why it's tricky to test this in integration/e2e?

Reply:
Added reasons in 617a33f5b